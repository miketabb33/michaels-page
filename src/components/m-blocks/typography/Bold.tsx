import styled from 'styled-components'
import { ChildrenProp } from '../../../types/ChildrenProp'
import React from 'react'

const BoldStyle = styled.strong`
  font-weight: 700;
  color: ${({ theme }) => theme.staticColor.blue_600};
`

const Bold = ({ children }: ChildrenProp) => <BoldStyle>{children}</BoldStyle>

export default Bold
