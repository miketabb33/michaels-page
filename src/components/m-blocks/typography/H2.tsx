import React from 'react'
import { ChildrenProp } from '../../../types/ChildrenProp'
import styled from 'styled-components'
import { Spacing, SpacingArgs, spacingController } from '../spacingController'

type StyledH2Props = {
  $spacing?: SpacingArgs
}

const H2Element = styled.h2<StyledH2Props>`
  font-size: 3rem;
  line-height: 5rem;
  font-weight: 600;
  color: ${({ theme }) => theme.color.text};
  ${({ $spacing }) => spacingController($spacing)}
`

const H2 = ({ children, spacing }: ChildrenProp & Spacing) => {
  return <H2Element $spacing={spacing}>{children}</H2Element>
}

export default H2
