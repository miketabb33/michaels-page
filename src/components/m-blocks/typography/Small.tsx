import styled, { css } from 'styled-components'
import { Spacing, SpacingArgs, spacingController } from '../spacingController'
import { ChildrenProp } from '../../../types/ChildrenProp'
import React from 'react'

type StyledSmallProps = {
  $spacing?: SpacingArgs
}

const SmallElement = styled.h3<StyledSmallProps>`
  ${({ $spacing }) => css`
    font-size: 1.2rem;
    font-weight: 900;
    line-height: 2rem;
    color: ${({ theme }) => theme.color.accent};

    ${spacingController($spacing)}
  `}
`

const Small = ({ children, spacing }: ChildrenProp & Spacing) => {
  return <SmallElement $spacing={spacing}>{children}</SmallElement>
}

export default Small
