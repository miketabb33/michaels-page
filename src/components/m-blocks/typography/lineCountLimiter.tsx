import { css } from 'styled-components'

export type LineLimitCount = {
  lineLimit?: number
}

export const lineCountLimiter = (lineLimit?: number) => {
  if (!lineLimit) return
  return css`
    display: -webkit-box;
    -webkit-line-clamp: ${lineLimit};
    -webkit-box-orient: vertical;
    overflow: hidden;
    text-overflow: ellipsis;
  `
}
