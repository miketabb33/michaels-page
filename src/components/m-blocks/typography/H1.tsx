import styled, { css } from 'styled-components'
import { SpacingArgs, spacingController } from '../spacingController'
import { lineCountLimiter } from './lineCountLimiter'

type H1Props = {
  ignoreDarkMode?: boolean
  $spacing?: SpacingArgs
  $lineLimit?: number
}

const H1 = styled.h1<H1Props>`
  ${({ theme, ignoreDarkMode, $spacing, $lineLimit }) => css`
    font-size: 3.6rem;
    line-height: 6rem;
    font-weight: 600;
    color: ${() => {
      const lightModeColor = theme.staticColor.blue_800
      if (ignoreDarkMode) return lightModeColor
      return theme.color.id === 'light' ? lightModeColor : theme.staticColor.blue_50
    }};
    ${spacingController($spacing)}
    ${lineCountLimiter($lineLimit)}
  `}
`

export default H1
