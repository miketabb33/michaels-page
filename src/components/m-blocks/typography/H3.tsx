import React from 'react'
import { ChildrenProp } from '../../../types/ChildrenProp'
import styled, { css } from 'styled-components'
import { Spacing, SpacingArgs, spacingController } from '../spacingController'
import { LineLimitCount, lineCountLimiter } from './lineCountLimiter'

type StyledH3Props = {
  $spacing?: SpacingArgs
  $lineLimit?: number
}

const H3Element = styled.h3<StyledH3Props>`
  ${({ theme, $spacing, $lineLimit }) => css`
    font-size: 2.4rem;
    line-height: 4.2rem;
    font-weight: 600;
    color: ${theme.color.text};
    ${spacingController($spacing)}
    ${lineCountLimiter($lineLimit)}
  `}
`

const H3 = ({ children, spacing, lineLimit }: ChildrenProp & Spacing & LineLimitCount) => {
  return (
    <H3Element $spacing={spacing} $lineLimit={lineLimit}>
      {children}
    </H3Element>
  )
}

export default H3
