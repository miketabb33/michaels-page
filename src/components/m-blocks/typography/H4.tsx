import styled, { css } from 'styled-components'
import { SpacingArgs, spacingController } from '../spacingController'
import { lineCountLimiter } from './lineCountLimiter'

type H4Props = {
  $spacing?: SpacingArgs
  $lineLimit?: number
}

const H4 = styled.h4<H4Props>`
  ${({ theme, $spacing, $lineLimit }) => css`
    font-size: 1.8rem;
    line-height: 3rem;
    font-weight: 700;
    color: ${theme.color.text};
    ${spacingController($spacing)}
    ${lineCountLimiter($lineLimit)}
  `}
`

export default H4
