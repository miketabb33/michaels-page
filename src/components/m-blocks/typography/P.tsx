import React from 'react'
import { ChildrenProp } from '../../../types/ChildrenProp'
import styled, { css } from 'styled-components'
import { Spacing, SpacingArgs, spacingController } from '../spacingController'
import { LineLimitCount, lineCountLimiter } from './lineCountLimiter'

type StyledPProps = {
  $spacing?: SpacingArgs
  $lineLimit?: number
}

const PElement = styled.p<StyledPProps>`
  ${({ theme, $spacing, $lineLimit }) => css`
    font-size: 1.8rem;
    line-height: 3rem;
    letter-spacing: 0.05rem;
    color: ${theme.color.text};
    ${spacingController($spacing)}
    ${lineCountLimiter($lineLimit)}
  `}
`
const P = ({ children, spacing, lineLimit }: ChildrenProp & Spacing & LineLimitCount) => {
  return (
    <PElement $spacing={spacing} $lineLimit={lineLimit}>
      {children}
    </PElement>
  )
}

export default P
