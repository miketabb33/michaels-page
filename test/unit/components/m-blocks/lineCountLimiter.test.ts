import { lineCountLimiter } from '../../../../src/components/m-blocks/typography/lineCountLimiter'

describe('Line Count Limiter', () => {
  it('should have 2', () => {
    expect(lineCountLimiter(2)).toMatchSnapshot()
  })
  it('should have 4', () => {
    expect(lineCountLimiter(4)).toMatchSnapshot()
  })
})
