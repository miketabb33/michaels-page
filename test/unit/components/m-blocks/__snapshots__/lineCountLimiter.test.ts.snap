// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Line Count Limiter should have 2 1`] = `
[
  "
    display: -webkit-box;
    -webkit-line-clamp: ",
  "2",
  ";
    -webkit-box-orient: vertical;
    overflow: hidden;
    text-overflow: ellipsis;
  ",
]
`;

exports[`Line Count Limiter should have 4 1`] = `
[
  "
    display: -webkit-box;
    -webkit-line-clamp: ",
  "4",
  ";
    -webkit-box-orient: vertical;
    overflow: hidden;
    text-overflow: ellipsis;
  ",
]
`;
